src/matcher.rs

```diff
@@ -118,22 +122,38 @@ impl StringMatcher {
         }
     }
 
-    fn parse_matcher_bare(chars: &mut ParsingIterator, bareword_end: char, anchor_start: bool) -> Self {
+    fn parse_matcher_bare<C>(chars: &mut ParsingIterator, bareword_end: C, anchor_start: bool) -> Self
+    where
+        C: Into<CharEnd>,
+    {
         let mut result = String::with_capacity(20); // just a guess
         let mut dropped = String::with_capacity(8); // also a guess
 
+        let bareword_end = bareword_end.into();
         let anchor_end = loop {
-            // Drop whitespace, but keep a record of it. If we see a char within this bareword (ie not end-of-input or
-            // the bareword_end), then we'll append that whitespace back.
-            chars.drop_to_while(&mut dropped, |ch| ch.is_whitespace());
-            let Some(ch) = chars.peek() else {
+            let ch = match bareword_end {
```

Comment:
This could probably be made more efficient by making CharEnd a trait, so that this if-else can be expanded at compile time. But it's not really worth it.